3-enrich/scala-common-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/registry/RefererParserEnrichmentSpec.scala

```diff
@@ -57,14 +62,45 @@ class ExtractRefererDetailsSpec extends Specification with DataTables {
       "Internal referer" !! "https://www.snowplowanalytics.com/account/profile" ! Medium.Internal ! None ! None |
       "Custom referer"   !! "https://www.internaldomain.com/path" ! Medium.Internal ! None ! None |
       "Unknown referer"  !! "http://www.spyfu.com/domain.aspx?d=3897225171967988459" ! Medium.Unknown ! None ! None |> {
-      (_, refererUri, medium, source, term) =>
-        RefererParserEnrichment(List("www.internaldomain.com"))
-          .extractRefererDetails(new URI(refererUri), PageHost) must_== Some(Referer(medium, source, term))
+      (_, refererUri, _, _, _) =>
+        val value: IO[Option[Referer]] = RefererParserEnrichment(List("www.internaldomain.com"), None)
+          .extractRefererDetails(new URI(refererUri), PageHost)
+          .getOrElse(None)
+        val p = value.unsafeRunSync()
+        p must beOneOf(
+          Some(InternalReferer),
+          Some(UnknownReferer),
+          Some(SocialReferer("Facebook")),
+          Some(EmailReferer("Yahoo! Mail")),
+          Some(SearchReferer("Google", Some("gateway oracle cards denise linn")))
+        )
     }
 
-  def e2 =
-    RefererParserEnrichment(List()).extractRefererDetails(
-      new URI("http://www.google.com/search?q=%0Agateway%09oracle%09cards%09denise%09linn&hl=en&client=safari"),
-      PageHost) must_== Some(
-      Referer(Medium.Search, Some("Google"), Some("gateway    oracle    cards    denise    linn")))
+  def e2 = {
+    val value: IO[Option[Referer]] = RefererParserEnrichment(List(), None)
+      .extractRefererDetails(
+        new URI("http://www.google.com/search?q=%0Agateway%09oracle%09cards%09denise%09linn&hl=en&client=safari"),
+        PageHost)
+      .getOrElse(None)
+    value.unsafeRunSync() must beSome(SearchReferer("Google", Some("gateway    oracle    cards    denise    linn")))
+  }
+
+  def e3 = {
+    val value: IO[Option[Referer]] = RefererParserEnrichment(List(), None, true)
+      .extractRefererDetails(new URI("android-app://m.facebook.com"), PageHost)
```

Comment:
Can you use enums for the hardcoded urls in case they change i n the future it would be easy to refactor